Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,295 @@
+                }
+
+                for (int j = 0; j < allAmounts.Length - 1; j++)
+                {
+                    try
+                    {
+                        var amount1 = decimal.Parse(allAmounts[j], NumberStyles.Currency | NumberStyles.Number);
+                        try
+                        {
+                            var amount2 = decimal.Parse(allAmounts[j + 1], NumberStyles.Currency | NumberStyles.Number);
+                            var lossForCurrentPerson = amount1 - amount2;
+
+                            //check if loss is greater then the current highest loss
+                            if (lossForCurrentPerson > HighestLossEver)
+                            {
+                                HighestLossEver = lossForCurrentPerson;
+                                PersonWithHighestLoss = currentPersonData[0];
+                            }
+                        }
+                        catch (FormatException)
+                        {
+                            Console.WriteLine($"{allAmounts[j + 1]} is not a  valid number");
+                            break;
+                        }
+                    }
+                    catch (FormatException)
+                    {
+                        Console.WriteLine($"{allAmounts[j]} is not a valid number");
+                        break;
+                    }
+                }
+            }
+            return $"{PersonWithHighestLoss} lost the most money. -¤{HighestLossEver}.";
         }
 
         /// <summary>
         /// Return name and current money of the richest person.
         /// </summary>
         public static string FindRichestPerson(string[] peopleAndBalances)
         {
-            return "";
+            if (IsInValidValidInput(peopleAndBalances))
+            {
+                return InValidOutput;
+            }
+
+            var richestPerson = new StringBuilder();
+            var highestBalance = FindRichest(peopleAndBalances, richestPerson);
+
+            var persons = richestPerson.ToString();
+
+            if (persons.Contains(','))
+            {
+                return $"{persons} are the richest people. ¤{highestBalance}.";
+            }
+            else
+            {
+                return $"{persons} is the richest person. ¤{highestBalance}.";
+            }
+        }
+
+        private static decimal FindRichest(string[] peopleAndBalances, StringBuilder RichestPerson)
+        {
+            var HighestBalance = decimal.MinValue;
+            for (int i = 0; i <= peopleAndBalances.Length - 1; i++)
+            {
+                var currentPersonData = peopleAndBalances[i].Split(',');
+                try
+                {
+                    var highestAmountOfPerson = decimal.Parse(currentPersonData[^1]);
+
+                    if (highestAmountOfPerson == HighestBalance)
+                    {
+                        if (i == peopleAndBalances.Length - 1)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use the '?:' operator here.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4967975)